src/utils/AutoDiscoveryUtils.js

```diff
@@ -68,19 +149,27 @@ export default class AutoDiscoveryUtils {
 
         const hsResult = discoveryResult['m.homeserver'];
         if (hsResult.state !== AutoDiscovery.SUCCESS) {
-            if (AutoDiscovery.ALL_ERRORS.indexOf(hsResult.error) !== -1) {
-                throw newTranslatableError(hsResult.error);
-            }
-            throw newTranslatableError(_td("Unexpected error resolving homeserver configuration"));
+            console.error("Error processing homeserver config:", hsResult);
+            if (!syntaxOnly || !AutoDiscoveryUtils.isLivelinessError(hsResult.error)) {
+                if (AutoDiscovery.ALL_ERRORS.indexOf(hsResult.error) !== -1) {
+                    throw newTranslatableError(hsResult.error);
+                }
+                throw newTranslatableError(_td("Unexpected error resolving homeserver configuration"));
+            } // else the error is not related to syntax - continue anyways.
         }
 
         const isResult = discoveryResult['m.identity_server'];
-        let preferredIdentityUrl = "https://vector.im";
+        let preferredIdentityUrl = "https://vector.im"; // We already know this is an IS, so don't validate it.
```

Comment:
Not sure what's going on with this comment as I can't see the code doing anything like this.

Reply:
There's a condition where the default value here gets passed straight through without ever hitting vector.im - this was what actually fixed a bug for a customer, as they could just remove `default_is_url` from their config and it magically works.

I can try and clarify the comment to say that though. Intention behind it is to say "please don't remove this feature when you're refactoring this code".